hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -479,4 +479,19 @@ private Map<TopicPartition, Long> getGroupOffsets(KafkaConsumer consumer, Set<To
     }
     return fromOffsets;
   }
+
+  /**
+   * get null safe kafka key from kafka key object
+   * @param kafkaKeyObject kafka key object
+   * @return kafka key
+   */
+  public static String getKafkaKey(Object kafkaKeyObject) {
+    String recordKey = null;
+    try {
+      recordKey = kafkaKeyObject.toString();
+    } catch (NullPointerException npe) {
```

Comment:
Why not just decide whether the `kafkaKeyObject` is null first?

Reply:
And the util method should not put here because the key has no relationship with kafka offset.

Reply:
Removed getKafkaKey util method from KafkaOffsetGen class & used `StringUtils#objToString` instead